R/visit.R

```diff
@@ -43,3 +43,166 @@ visit_one <- function(pd_flat, funs) {
   reduce(funs, function(x, fun) fun(x),
          .init = pd_flat)
 }
+
+
+#' Propagate context to terminals
+#'
+#' Implements a very specific pre-visiting scheme, namely to propagate
+#'   indention, spaces and lag_newlines to inner token to terminals. This means
+#'   that information regarding indention, linebreaks and spaces (which is
+#'   relative in `pd_nested`) will be converted into absolute.
+#' @inherit context_towards_terminals
+#' @seealso context_towards_terminals visitors
+context_to_terminals <- function(pd_nested,
```

Comment:
How about:

```r
visit_context_to_terminals <- function(pd) {
  pd <- context_towards_terminals(pd, pd$outer_lag_newlines, ...)
  pd$child <- map2(pd$child, ..., function(x, y) { x[["outer_lag_newlines"]] <- y})
  ...
  pd
}
```

This stores the information for the next stage of the visitor in the children, where it is then picked up as needed.